src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.History/Properties/Resources.resx

```diff
@@ -0,0 +1,129 @@
+    <xsd:import namespace="http://www.w3.org/XML/1998/namespace" />
+    <xsd:element name="root" msdata:IsDataSet="true">
+      <xsd:complexType>
+        <xsd:choice maxOccurs="unbounded">
+          <xsd:element name="metadata">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" />
+              </xsd:sequence>
+              <xsd:attribute name="name" use="required" type="xsd:string" />
+              <xsd:attribute name="type" type="xsd:string" />
+              <xsd:attribute name="mimetype" type="xsd:string" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="assembly">
+            <xsd:complexType>
+              <xsd:attribute name="alias" type="xsd:string" />
+              <xsd:attribute name="name" type="xsd:string" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="data">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+                <xsd:element name="comment" type="xsd:string" minOccurs="0" msdata:Ordinal="2" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" msdata:Ordinal="1" />
+              <xsd:attribute name="type" type="xsd:string" msdata:Ordinal="3" />
+              <xsd:attribute name="mimetype" type="xsd:string" msdata:Ordinal="4" />
+              <xsd:attribute ref="xml:space" />
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="wox_plugin_history_plugin_name" xml:space="preserve">
+    <value>History</value>
+  </data>
+  <data name="wox_plugin_history_plugin_description" xml:space="preserve">
+    <value>Quick access to previous selected results.</value>
```

Comment:
```suggestion
    <value>Quick access to previous searched results.</value>
```

Reply:
Hmm I think a name change is the correct option since "Quick access to previous **selected** results" is the correct and intended effect. 

> 1. The history entry is only created/working if you select an item after entering a search term...

100% correct. We don't want to have the history "cluttered" with everything that ever showed up. The main reason this won't work for us, but does for "search engines" is that we don't have a "search" button. So as users are typing it will show many-many things they don't care about. Since PT Run is designed for real-time results, those results are not as valuable as the one the users actually select/click on.

For example, if the user is fumbling to get a "web search" or "calculator" plugin query correct, then select/clicks the final correct result, that is the one we want in the history. Not the ones with all the typos. 

Please explain because it seems to do what it says it does. 

The current description "Quick access to previous **selected** results" already makes that clear. 

Is it the name of the plugin that needs to change? I played with "Run History" in the past and it didn't seem right. 

> 2. I found the reason for the build problems...

Sadly, yes, this is known. I replied on this issue above, with possible solutions. Only my 2nd solution "reflection" is one that can solve this without code changes/refactoring outside this project. I am not comfortable doing the larger refactor without a "go ahead" before hand. But I also really-really don't like that options. 

If we do nothing, the only issue is what you stated, "you can't debug PowerLauncher including all Plugins directly..." I don't see a obvious solution without down side. Also, while this is a fair "issue" it will only ever affect the devs that are making changes to the History plugin. For them (you and me), it's required to build the History plugin manually. 

Which of the suggested solutions, or another, do you like?


@htcfreek thank you very much for the review. I feel this is moving forward and I love this plugin. 

I don't have don't have any changes to make based on this review so can you provide feedback on my comments above?

Reply:
See https://github.com/microsoft/PowerToys/pull/19569#issuecomment-1214382799 for my answers.